jsonschema-core.xml

```diff
@@ -1256,32 +1252,79 @@
-                        resource intended for use as a meta-schema.  It MUST NOT appear in subschemas.
-                    </t>
-                    <t>
-                        The "$vocabulary" keyword MUST be ignored in schema resources that
-                        are not being processed as a meta-schema.  This allows validating
-                        a meta-schema M against its own meta-schema M' without requiring
-                        the validator to understand the vocabularies declared by M.
-                    </t>
-                    <section title="Default vocabularies">
+                    <section title="Required, optional, and omitted vocabularies">
+                        <t>
+                            A schema is said to use a dialect and its constituent vocabularies if it is
+                            associated with a meta-schema defining the dialect with "$vocabulary",
+                            either through "$schema", through appropriately defined media type parameters
+                            or link relation types, or through documented default implementation-defined
+                            behavior in the absence of an explicit meta-schema.  If a meta-schema
+                            does not contain "$vocabulary", the set of vocabularies in use is determined
+                            according to section <xref target="default-vocabs" format="counter"></xref>.
+                        </t>
+                        <t>
+                            Any vocabulary in use by a schema and understood by the implementation
+                            MUST be processed in a manner consistent with the semantic definitions
+                            contained within the vocabulary, regardless of whether that vocabulary
+                            is required or optional.
+                        </t>
+                        <t>
+                            Any vocabulary that is not present in "$vocabulary" MUST NOT be made
+                            available for use in schemas described by that meta-schema, except for
+                            the core vocabulary as specified by the introduction to section
+                            <xref target="core" format="counter"></xref>.
+                        </t>
+                        <t>
+                            Implementations that do not support a vocabulary required by a schema
+                            MUST refuse to process that schema.
+                        </t>
+                        <t>
+                            Implementations that do not support a vocabulary that is optionally used
+                            by a schema SHOULD proceed with processing the schema.  The keywords will
+                            be considered to be unrecognized keywords as addressed by section
+                            <xref target="unrecognized" format="counter"></xref>.  Note that since
```

Comment:
This is fine for now, but may change with the outcome of [Proposal: Disallow non-vocabulary keywords](https://github.com/orgs/json-schema-org/discussions/241).

Reply:
yes, definitely, and I'll be happy to further update it.  I did not want to assume the outcome of a complex topic.

Reply:
No worries.  Just wanted to have a link and state that this is still open discussion.